lib/postgrex/protocol.ex

```diff
@@ -2305,7 +2309,9 @@ defmodule Postgrex.Protocol do
         recv_execute(s, status, query, rows, buffer)
 
       {:disconnect, _, _} = dis ->
-        dis
+        with {_, %{reason: :closed} = err, s} <- dis do
+          {:disconnect, %{err | reason: :execute_closed}, s}
+        end
```

Comment:
Do we still need this one? Its existence still means we are being too broad in a retry somewhere.